shell/platform/linux/fl_engine.cc

```diff
@@ -901,6 +908,115 @@ void fl_engine_send_mouse_pointer_event(FlEngine* self,
   self->embedder_api.SendPointerEvent(self->engine, &fl_event, 1);
 }
 
+// Set's |event_data|'s phase to either kMove or kHover depending on the current
+// primary mouse button state.
+void SetEventPhaseFromCursorButtonState(FlEngine* self,
+                                        FlutterPointerEvent* event_data,
+                                        const PointerState* state) {
+  // For details about this logic, see FlutterPointerPhase in the embedder.h
+  // file.
+  if (state->buttons == 0) {
+    event_data->phase = state->flutter_state_is_down
+                            ? FlutterPointerPhase::kUp
+                            : FlutterPointerPhase::kHover;
+  } else {
+    event_data->phase = state->flutter_state_is_down
+                            ? FlutterPointerPhase::kMove
+                            : FlutterPointerPhase::kDown;
+  }
+}
+
+void fl_engine_send_pointer_event(FlEngine* self,
```

Comment:
The engine has methods for each type of event, e.g. `fl_engine_send_pointer_pan_zoom_event` - adding this general event method doesn't fit in with the existing architecture. How many new pointer events are required? Could there be specific methods for these?

Reply:
I feel the naming could be clearer. It might make more sense as fl_engine_send_touch_pointer_event. Otherwise, if I'm getting this right, we’d end up needing five separate methods for each phase we use. Should I go ahead with that approach?

Reply:
I would say yes, make five methods to match the existing code. We can always consolidate them later if necessary.